components/Elevation/src/UIColor+MaterialElevation.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <CoreGraphics/CoreGraphics.h>
+#import <UIKit/UIKit.h>
+
+/**
+ Provides extension to UIColor for Material Elevation usage.
+ */
+@interface UIColor (MaterialElevation)
+
+/**
+ Returns a color that takes the specified elevation value into account.
+ The color is the blended color of Surface and Elevation Overlay in
+ https://material.io/design/color/dark-theme.html#properties
+ @param elevation The mdc_absoluteElevation value to use when resolving the color.
+ */
+- (nonnull UIColor *)mdc_resolvedColorWithElevation:(CGFloat)elevation;
+
+/**
+ Returns a color that takes the specified elevation value and traits into account.
+ When userInterfaceStyle is UIUserInterfaceStyleDark in traitCollection, elevation will be used
+ to resolve the color.
+ Negative elevation is treated as 0. Pattern-based UIColor is not supported.
+ UIColor in UIExtendedGrayColorSpace will be resolved to UIExtendedSRGBColorSpace.
```

Comment:
Please mention pattern-based colors not supported on both APIs.

Reply:
Addressed in 1a7be2d484ec059878b1b5a310b9c03bb94e28b1